### ISSUE TITLE
Documention: Please add information about migrating

### ISSUE THREAD

**Issue by rvanlaar** [carltongibson/django-filter#1174]
I'm migrating from 1.1.0 to 2.2.0.

Please add information about:
- how to implement the LookupChoiceFilter, as it's now clear to me how that works and is lacking an example
- how to handle the old filter_class class variables, as it's not even referenced in the documentation anymore.

If you really want to make my day:
- Release a new 1.1.X version with proper deprecation warnings for all deprecations in 2.2

**Comment by rpkilby** [comment]
Hi @rvanlaar. Have you seen the [migration guide](https://django-filter.readthedocs.io/en/master/guide/migration.html)? It discusses both of these issues.

**Comment by rvanlaar** [comment]
@rpkilby Yes I have. I wouldn't have opened this issue otherwise. I could ask you the same thing about this issue. I won't.

What would make it more clear to you what I meant? 

For the LookupChoiceFilter, I had to convert a `_FIELDTYPE_Filter` to a LookupChoiceFilter. 
Not just the `lookup_expr` was renamed, as was the case with the `name` to `field_name`. 
The whole function in which `lookup_expr` was used had to change. 
That was not clear to me by reading the migration guide.

**Comment by rvanlaar** [comment]
@rpkilby Yes I have. I wouldn't have opened this issue otherwise. I could ask you the same thing about this issue. I won't.

What would make it more clear to you what I meant? 

For the LookupChoiceFilter, I had to convert a `_FIELDTYPE_Filter` to a LookupChoiceFilter. 
Not just the `lookup_expr` was renamed, as was the case with the `name` to `field_name`. 
The whole function in which `lookup_expr` was used had to change. 
That was not clear to me by reading the migration guide.

**Comment by rpkilby** [comment]
> Just to be aware, it doesn't show up in the search results.

Good to know. I'm not sure how RTD's search works, but searching for `filter_class` doesn't return the migration guide, but searching for `ViewSet.filter_class` does. Maybe RTD doesn't do partial word matches?

In regards to the `lookup_expr` changes, what other information would be helpful to you? The migration docs state that the feature has been removed in favor of `LookupChoiceFilter`, and the linked docs provide an example.

**Comment by rvanlaar** [comment]
> In regards to the lookup_expr changes, what other information would be helpful to you? The migration docs state that the feature has been removed in favor of LookupChoiceFilter, and the linked docs provide an example.

It would've been nice if this was mentioned as a deprecation warning in the previous release. I'm not even sure why there actually is a section about the warnings if they're not even used:
https://django-filter.readthedocs.io/en/master/guide/migration.html#enabling-warnings

This is what the docs say:
> The Filter.lookup_expr argument no longer accepts None or a list of expressions. Use the LookupChoiceFilter instead.

Something like: Filters don't support lookup_expr any more. If you need an old lookup_expr migrate your Filter to a LookupChoiceFilter.

For example: `DateFilter(name="created_date", lookup_expr=['gte', 'lte'])` is replaced by:
`LookupChoiceFilter(field_name="created_date", field_class=forms.DataField, lookup_choices=[('gt', 'Greater than'), ('lt', 'Less than')]`

**Comment by rpkilby** [comment]
> It would've been nice if this was mentioned as a deprecation warning in the previous release. 

We actually raise an explicit assertion instead of just a warning for this. 
https://github.com/carltongibson/django-filter/blob/400469f03f7ebac6e0a7d8cbe6a7706f71e14c96/django_filters/filters.py#L85-L88

In regards to example usage, the migration docs do link to the `LookupChoiceFilter`, which provides this example.

**Comment by rvanlaar** [comment]
We seem to hit mismatch after mismatch in our communication in this thread. It feels like you're not seeing the forest from the trees.

I've raised a few issues about django-filter that could make it better.
Things that weren't clear to me while dealing with the upgrade.

It would be nice if @carltongibson can look it over and decide if the docs and deprecation process needs improvement.
